packages/react/dialog/src/Dialog.tsx

```diff
@@ -169,7 +177,15 @@ const DialogOverlayImpl = React.forwardRef<DialogOverlayImplElement, DialogOverl
     const { __scopeDialog, ...overlayProps } = props;
     const context = useDialogContext(OVERLAY_NAME, __scopeDialog);
     return (
-      <Primitive.div data-state={getState(context.open)} {...overlayProps} ref={forwardedRef} />
+      <RemoveScroll as={Slot} allowPinchZoom={context.allowPinchZoom}>
+        <Primitive.div
+          data-state={getState(context.open)}
+          {...overlayProps}
+          ref={forwardedRef}
+          // We re-enable pointerEvents on the Overlay to enable scrolling
```

Comment:
This comment tripped me up because I thought:

> But `RemoveScroll` should only disable pointer events outside of overlay so why do we need to re-enable them?

.... then I remembered your comment in the PR description. Would you mind if we made that a little clearer somehow? Something like:

```suggestion
          // We re-enable pointer-events prevented by `Dialog.Content` to allow scrolling the overlay.
```